service/frontend/fx.go

```diff
@@ -599,6 +601,7 @@ func HandlerProvider(
 	healthServer *health.Server,
 	membershipMonitor membership.Monitor,
 	healthInterceptor *interceptor.HealthInterceptor,
+	scheduleSpecBuilder *scheduler.SpecBuilder,
```

Comment:
Hi @dnr,
Quick question. What is the benefit of having a pointer to scheduleSpecBuilder over value?

Reply:
Yes, the goal is to share it across the whole process. It just has a cache (in the future possibly multiple caches) and there's no harm in sharing it widely